sdk/core/Azure.Core/src/Operation.cs

```diff
@@ -25,8 +25,33 @@ public abstract class Operation
         /// <param name="rehydrationToken">The rehydration token.</param>
         /// <param name="options">The client options.</param>
         /// <returns>The long-running operation.</returns>
-        public static Operation<T> Rehydrate<T>(HttpPipeline pipeline, RehydrationToken? rehydrationToken, ClientOptions? options = null) where T : IPersistableModel<T>
-            => new RehydrationOperation<T>(pipeline, rehydrationToken, options);
+        public static Operation<T> Rehydrate<T>(HttpPipeline pipeline, RehydrationToken rehydrationToken, ClientOptions? options = null) where T : IPersistableModel<T>
+        {
+            Argument.AssertNotNull(pipeline, nameof(pipeline));
+            Argument.AssertNotNull(rehydrationToken, nameof(rehydrationToken));
+
+            IOperationSource<T> source = new GenericOperationSource<T>();
+            var nextLinkOperation = (NextLinkOperationImplementation)NextLinkOperationImplementation.Create(pipeline, rehydrationToken);
+            var operation = NextLinkOperationImplementation.Create(source, nextLinkOperation);
+            var operationState = operation.UpdateStateAsync(false, default).EnsureCompleted();
+            return new RehydrationOperation<T>(nextLinkOperation, operationState, operation, options);
+        }
+
+        /// <summary>
+        /// Rehydrates an operation from a <see cref="RehydrationToken"/>.
+        /// </summary>
+        /// <param name="pipeline">The Http pipeline.</param>
+        /// <param name="rehydrationToken">The rehydration token.</param>
+        /// <param name="options">The client options.</param>
+        /// <returns>The long-running operation.</returns>
+        public static Operation Rehydrate(HttpPipeline pipeline, RehydrationToken rehydrationToken, ClientOptions? options = null)
+        {
+            Argument.AssertNotNull(pipeline, nameof(pipeline));
+            Argument.AssertNotNull(rehydrationToken, nameof(rehydrationToken));
+            var nextLinkOperation = (NextLinkOperationImplementation)NextLinkOperationImplementation.Create(pipeline, rehydrationToken);
+            var operationState = nextLinkOperation.UpdateStateAsync(false, default).EnsureCompleted();
```

Comment:
Same comment as https://github.com/Azure/azure-sdk-for-net/pull/42686/files/adc92770313b373c0e0ed3afd642114d21b1b967..6e2eb4a965f3d33fc115de40186cb318bbd81677#r1539497852